azure-pipelines.yml

```diff
@@ -51,6 +51,17 @@ steps:
   workingDirectory: $(Pipeline.Workspace)/target/python-wheels/
   displayName: 'Install Python dependencies'
 
+- script: |
+    set -ex
+    # Install .NET CORE
+    #wget https://packages.microsoft.com/config/ubuntu/20.04/packages-microsoft-prod.deb -O packages-microsoft-prod.deb
+    #sudo dpkg -i packages-microsoft-prod.deb
+    curl -sSL https://packages.microsoft.com/keys/microsoft.asc | sudo apt-key add -
+    sudo apt-add-repository https://packages.microsoft.com/debian/10/prod
+    sudo apt-get update
+    sudo apt-get install -y dotnet-sdk-5.0
```

Comment:
Install is failing due to unmet dependencies. Try this:

```
sudo apt-get install -y dotnet-sdk-5.0 || sudo apt-get -y install -f
```

Reply:
with sudo apt-get -y install -f
```
with crul -sSL and debian 10，no error but also no color bar: https://dev.azure.com/mssonic/build/_build/results?buildId=21689&view=codecoverage-tab
with curl -sSL and ubuntu 20 , exits error: https://dev.azure.com/mssonic/build/_build/results?buildId=21692&view=logs&j=12f1170f-54f2-53f3-20dd-22fc7dff55f9
with wget and ubuntu 20, no error but also no clolor bar: https://dev.azure.com/mssonic/build/_build/results?buildId=21694&view=codecoverage-tab
with wget and debian 10, not test.
```

sudo apt --fix-broken install -y
```
with crul -sSL and debian 10, no error and exits color bar: https://dev.azure.com/mssonic/build/_build/results?buildId=21696&view=codecoverage-tab
```

Reply:
Those commands should be equivalent. I think the difference here is that you're now calling `sudo apt --fix-broken install -y` before installing the package, rather than if the package install failed. Can you try replacing that line with `sudo apt-get -y install -f` and see if it still works? I prefer `apt-get` to `apt`. See the warning message, `WARNING: apt does not have a stable CLI interface. Use with caution in scripts.`